sqlglot/dialects/clickhouse.py

```diff
@@ -756,6 +756,26 @@ def _parse_projection_def(self) -> t.Optional[exp.ProjectionDef]:
         def _parse_constraint(self) -> t.Optional[exp.Expression]:
             return super()._parse_constraint() or self._parse_projection_def()
 
+        def _parse_alias(
+            self, this: t.Optional[exp.Expression], explicit: bool = False
+        ) -> t.Optional[exp.Expression]:
+            # In clickhouse "SELECT <expr> APPLY(...)" is a query modifier,
+            # so "APPLY" shouldn't be parsed as <expr>'s alias. However, "SELECT <expr> apply" is a valid alias
+            if self._match_pair(TokenType.APPLY, TokenType.L_PAREN, advance=False):
+                return this
+
+            return super()._parse_alias(this=this, explicit=explicit)
+
+        def _parse_projections(self) -> t.List[exp.Expression]:
+            exprs = self._parse_expressions()
+
+            # Clickhouse allows "SELECT <expr> APPLY(func) [...,]" modifier
+            while self._match_pair(TokenType.APPLY, TokenType.L_PAREN):
+                exprs[-1] = exp.Apply(this=exprs[-1], expression=self._parse_var(any_token=True))
```

Comment:
I'm not sure if this is the right way to set this, because the `exp.Apply` won't get its `index` set. Let's instead use the `set` method and specify the index. Can we have >1 projections _and_ an `APPLY` operator? We can try to simplify, e.g. by always setting at index 0 or something if there can only be 1 projection.

Regarding `_parse_var` - can you have namespaced functions? E.g. `schema.func` won't be parsed here.

Reply:
Both errors re: namespaced functions seem reasonable, I don't see something in the messages that leads me to believe it's unsupported. Maybe what you tried wasn't right?

Re: `APPLY` and projections - you don't need `_parse_projection`, you could just override `_parse_expression` for CH.